From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 27 Oct 2020 20:16:39 +0000
Subject: screen: Ignore some more environment variables

- `GJS_DEBUG_OUTPUT` and `GJS_DEBUG_TOPICS` are set by GNOME Shell.
- `G_ENABLE_DIAGNOSTIC` is set by GNOME Terminal itself.

Bug: https://gitlab.gnome.org/GNOME/gnome-terminal/-/issues/303
Bug-Debian: https://bugs.debian.org/973918
Applied-upstream: no
---
 src/terminal-screen.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/terminal-screen.c b/src/terminal-screen.c
index 8faca97..0303a03 100644
--- a/src/terminal-screen.c
+++ b/src/terminal-screen.c
@@ -1460,7 +1460,10 @@ terminal_screen_get_child_environment (TerminalScreen *screen,
   g_hash_table_remove (env_table, "EXIT_STATUS");
   g_hash_table_remove (env_table, "GIO_LAUNCHED_DESKTOP_FILE");
   g_hash_table_remove (env_table, "GIO_LAUNCHED_DESKTOP_FILE_PID");
+  g_hash_table_remove (env_table, "GJS_DEBUG_OUTPUT");
+  g_hash_table_remove (env_table, "GJS_DEBUG_TOPICS");
   g_hash_table_remove (env_table, "GNOME_DESKTOP_ICON");
+  g_hash_table_remove (env_table, "G_ENABLE_DIAGNOSTIC");
   g_hash_table_remove (env_table, "INVOCATION_ID");
   g_hash_table_remove (env_table, "JOURNAL_STREAM");
   g_hash_table_remove (env_table, "LINES");
